samples/MQTTSN/src/paho_iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,579 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#include "azure/iot/az_iot_hub_client.h"
+#include "src/MQTTSNPacket.h"
+#include "transport.h"
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// DO NOT MODIFY: IoT Hub Hostname Environment Variable Name
+#define ENV_IOT_HUB_HOSTNAME "AZ_IOT_HUB_HOSTNAME"
+
+// #define MQTT_QOS 0 // if not defined, default qos is 1
+#define TELEMETRY_SEND_INTERVAL 1 // seconds
+#define NUMBER_OF_MESSAGES 5
+#define TELEMETRY_PAYLOAD \
+  "{\"d\":{\"myName\":\"IoT mbed\",\"accelX\":12,\"accelY\":4,\"accelZ\":12,\"temp\":18}}"
+
+#if !defined(MQTT_QOS) || (defined(MQTT_QOS) && MQTT_QOS == 1) // default to qos 1
+#define ENABLE_PUBACK true
+#endif

```

Comment:
how about this?

#ifdef AZ_TELEMERTY_QOS0
#undef ENABLE_PUBACK
#else 
#define ENABLE_PUBACK
#endif